### PR TITLE
MM-47838 - Conclude A/B test for upgrade button

### PR DESCRIPTION
**Pull request by ConorMacpherson**
#### Summary

A/B testing was setup to see if the text "View Plans" could influence/drive greater traffic to the plans modals.
Based on the [data](https://mattermost.looker.com/explore/data_warehouse/feature_flag_telemetry?toggle=fil&qid=Yf0VTKJdfGC5lro9H7Io4z), "View Plans" performed better than "Upgrade"

This change removes references to the feature flag made for this A/B testing along with the "Upgrade" text for this CTA.

#### Ticket Link

[MM-47838](https://mattermost.atlassian.net/browse/MM-47838)

#### Related Pull Requests

https://github.com/mattermost/mattermost-server/pull/21508

#### Screenshots

|  before (A/B)  |  after  |
|----|----|
|Screen Shot 2022-10-25 at 2 11 56 PM  | Screen Shot 2022-10-25 at 2 13 35 PM |
| Screen Shot 2022-10-25 at 2 12 47 PM |  |

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by ConorMacpherson** [comment]
Just waiting to be added to the completed CLA agreement list.

**Review by neallred — CHANGES_REQUESTED** [review]
What's changed so far looks good.
There's another spot we can remove code:
https://github.com/mattermost/mattermost-webapp/blob/master/e2e/playwright/support/server/default_config.ts#L671
And there's some e2e tests that need to be updated in `e2e/cypress/tests/integration/enterprise/cloud/billing/self_hosted_pricing_modal_spec.js`: ones that select `#UpgradeButton` and assert on containing "Upgrade" text need to be updated to assert on "View plans" text.

**Comment by ConorMacpherson** [comment]
Thanks for pointing those out Nathaniel! I didn't notice them before, but I'll remove those other references :)

**Comment by ConorMacpherson** [comment]
e2e tests updated :)

File: e2e/cypress/tests/integration/enterprise/cloud/billing/self_hosted_pricing_modal_spec.js [hunk 1]
```diff
@@ -162,7 +162,7 @@ describe('Self hosted Pricing modal', () => {
         // * Check that professional upgrade button is available
         cy.get('#pricingModal').should('be.visible');
         cy.get('#professional').should('be.visible');
-        cy.get('#professional_action').should('not.be.disabled').contains('Upgrade');
```
Comment (neallred) [inline e2e/cypress/tests/integration/enterprise/cloud/billing/self_hosted_pricing_modal_spec.js:165]: I thought all these needed upgrading at first, but reading more closely, looks like only the first one (#UpgradeButton) needs to be update. The other ones are for a #professional_action, which didn't change.

Comment (ConorMacpherson) [reply e2e/cypress/tests/integration/enterprise/cloud/billing/self_hosted_pricing_modal_spec.js:165]: Oh, my bad. I should be paying more attention. I just did a search and replace.

**Review by neallred — APPROVED** [review]
LGTM now, thanks @ConorMacpherson !

**Comment by ConorMacpherson** [comment]
Thank you @neallred ! :)

**Comment by ConorMacpherson** [comment]
Should be merged **AFTER** October 27th, 2022 to get into v7.5

**Review by AGMETEOR — APPROVED** [review]

**Comment by yasserfaraazkhan** [comment]
Thank you Conor, 
the tests have passed and even manually nothing is broken.

**Pull request merged**